### PR TITLE
[droid] Add leanback recommendation and search

### PR DESCRIPTION
**Pull request by koying**
## Description
Add leanback recommendation and search + depending changes

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by koying** [comment]
Seems github is confused by the dates ;)

jenkins build this please

File: cmake/scripts/common/Macros.cmake [hunk 1]
```diff
@@ -615,13 +616,14 @@ macro(core_find_versions)
   include(CMakeParseArguments)
   core_file_read_filtered(version_list ${CORE_SOURCE_DIR}/version.txt)
   string(REPLACE " " ";" version_list "${version_list}")
-  cmake_parse_arguments(APP "" "APP_NAME;COMPANY_NAME;WEBSITE;VERSION_MAJOR;VERSION_MINOR;VERSION_TAG;VERSION_CODE;ADDON_API" "" ${version_list})
+  cmake_parse_arguments(APP "" "APP_NAME;COMPANY_NAME;WEBSITE;VERSION_MAJOR;VERSION_MINOR;VERSION_TAG;VERSION_CODE;ADDON_APIi;APP_PACKAGE" "" ${version_list})
```
Comment (wsnipex) [inline cmake/scripts/common/Macros.cmake]: ADDON_APIi

Comment (koying) [inline cmake/scripts/common/Macros.cmake]: vi ;)

File: xbmc/PlayListPlayer.cpp [hunk 2]
```diff
@@ -279,6 +280,9 @@ bool CPlayListPlayer::Play(int iSong, std::string player, bool bAutoPlay /* = fa
 
   m_iCurrentSong = iSong;
   CFileItemPtr item = playlist[m_iCurrentSong];
+  if (item->IsVideoDb() && !item->HasVideoInfoTag())
```
Comment (da-anda) [inline xbmc/PlayListPlayer.cpp]: I'm not sure this kind of logic should be something the PlaylistPlayer has to care about. It's IMO not within it's domain. Can't this be handled elsewhere?

File: xbmc/utils/URIUtils.cpp [hunk 3]
```diff
@@ -215,6 +215,17 @@ void URIUtils::Split(const std::string& strFileNameAndPath,
   strPath = strFileNameAndPath.substr(0, i+1);
   // everything to the right of the directory separator
   strFileName = strFileNameAndPath.substr(i+1);
+
+  // ignore options
```
Comment (Razzeee) [inline xbmc/utils/URIUtils.cpp]: You might want to pick this test https://github.com/xbmc/xbmc/pull/10134/files#diff-c2c11e94617bec636884b0ccdf9dbaf6R125

File: xbmc/video/windows/GUIWindowVideoNav.cpp [hunk 4]
```diff
@@ -131,9 +132,50 @@ bool CGUIWindowVideoNav::OnMessage(CGUIMessage& message)
...
+              OnItemInfo(*pItem, scrapper);
+            }
+            break;
+          }
+        }
+        if (i >= m_vecItems->Size() && url.GetOption("showinfo") == "true")
+        {
+          // We are here if the item is filtered out in the nav window
+          std::string path = message.GetStringParam(0);
+          CFileItem* item = new CFileItem(path, URIUtils::HasSlashAtEnd(path));
```
Comment (tamland) [inline xbmc/video/windows/GUIWindowVideoNav.cpp]: leaks..

**Comment by MartijnKaijser** [comment]
Works fine

**Comment by koying** [comment]
If no-one else has comments, I'll merge tomorrow

**Comment by koying** [comment]
jenkins build and merge

File: xbmc/filesystem/VideoDatabaseFile.cpp [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+{
+  std::string strPath = URIUtils::GetDirectory(url.Get());
+  if (strPath.empty())
+    return VIDEODB_CONTENT_UNKNOWN;
+
+  std::vector<std::string> pathElem = StringUtils::Split(strPath, "/");
+  if (pathElem.size() == 0)
+    return VIDEODB_CONTENT_UNKNOWN;
+
+  std::string itemType = pathElem.at(2);
```
Comment (fritsch) [inline xbmc/filesystem/VideoDatabaseFile.cpp]: You need to check pathElem's length before accessing it at position 2.

File: xbmc/playlists/PlayList.cpp [hunk 6]
```diff
@@ -95,10 +95,6 @@ void CPlayList::Add(const CFileItemPtr &item, int iPosition, int iOrder)
   else
     item->m_iprogramCount = iOrder;
 
-  // videodb files are not supported by the filesystem as yet
-  if (item->IsVideoDb())
-    item->SetPath(item->GetVideoInfoTag()->m_strFileNameAndPath);
-
```
Comment (fritsch) [inline xbmc/playlists/PlayList.cpp]: You need to workaround the cases here, when VideoDataBaseFS fails and fall back to the item->SetPath way of doing it.

**Comment by mediaminister** [comment]
For "item selection via ActivateWindow", but I don't understand yet what that means. Maybe @koying can explain.

**Pull request merged**